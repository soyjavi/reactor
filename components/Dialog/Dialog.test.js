import React from 'react';
import renderer from 'react-test-renderer';

import { Dialog } from '..';
import MOCKS from './Dialog.mocks';

const position = 'bottom';
const fn = () => {};

jest.useFakeTimers();

describe('<Dialog>', () => {
  it('renders', () => {
    const tree = renderer.create(<Dialog />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {visible}', () => {
    const tree = renderer.create(<Dialog visible />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {children}', () => {
    const tree = renderer.create(<Dialog>{MOCKS.CHILDREN}</Dialog>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {highlight}', () => {
    const tree = renderer.create(<Dialog title={MOCKS.TITLE} highlight />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onClose}', () => {
    const tree = renderer.create(<Dialog onClose={fn} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {position}', () => {
    const tree = renderer.create(<Dialog position={MOCKS.POSITION} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Dialog style={MOCKS.STYLE} styleOverlay={MOCKS.STYLE_CONTAINER} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('hook:styler', () => {
    const snapshot = renderer.create(<Dialog marginBottom="M" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
