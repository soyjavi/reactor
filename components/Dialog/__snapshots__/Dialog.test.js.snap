// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog> renders 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {background} 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {children} 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4a4a4a",
                      "fontFamily": undefined,
                      "textAlignVertical": "center",
                    },
                    Object {
                      "fontSize": 16,
                      "lineHeight": 19.2,
                    },
                    false,
                    Object {},
                  ]
                }
              >
                lorem ipsum...
              </Text>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {highlight} 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#4a4a4a",
                    "fontFamily": undefined,
                    "textAlignVertical": "center",
                  },
                  Object {
                    "fontSize": 18,
                  },
                  false,
                  Object {
                    "color": "#ffffff",
                    "flex": 1,
                    "fontWeight": "700",
                    "paddingHorizontal": 8,
                  },
                ]
              }
            >
              Lorem ipsum...
            </Text>
            <View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                style={
                  Array [
                    undefined,
                    Array [
                      Object {
                        "borderRadius": 3,
                      },
                      Object {
                        "borderRadius": 26,
                      },
                      false,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignContent": "center",
                        "alignItems": "center",
                        "borderRadius": 3,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 44,
                        "minWidth": 44,
                        "paddingHorizontal": 28,
                      },
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                      },
                      false,
                      false,
                      false,
                      Object {
                        "borderRadius": 26,
                      },
                      Object {
                        "paddingHorizontal": 0,
                      },
                      false,
                      false,
                      false,
                      false,
                      false,
                    ]
                  }
                >
                  <Text />
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "flexDirection": "row",
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderRadius": 3,
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 22,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#ffffff",
                        "height": 0,
                        "opacity": 0,
                        "width": 0,
                        "zIndex": 0,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <RCTScrollView
            onScroll={[Function]}
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {onClose} 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                style={
                  Array [
                    undefined,
                    Array [
                      Object {
                        "borderRadius": 3,
                      },
                      Object {
                        "borderRadius": 26,
                      },
                      false,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignContent": "center",
                        "alignItems": "center",
                        "borderRadius": 3,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 44,
                        "minWidth": 44,
                        "paddingHorizontal": 28,
                      },
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                      },
                      false,
                      false,
                      false,
                      Object {
                        "borderRadius": 26,
                      },
                      Object {
                        "paddingHorizontal": 0,
                      },
                      false,
                      false,
                      false,
                      false,
                      false,
                    ]
                  }
                >
                  <Text />
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "flexDirection": "row",
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderRadius": 3,
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 22,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#4a4a4a",
                        "height": 0,
                        "opacity": 0,
                        "width": 0,
                        "zIndex": 0,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {reverse} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {style} 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Object {
                "backgroundColor": "#10c0bc",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {styleContainer} 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Object {
          "backgroundColor": "#f44336",
          "height": "50%",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {title} 1`] = `
<View
  pointerEvents="none"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#4a4a4a",
                    "fontFamily": undefined,
                    "textAlignVertical": "center",
                  },
                  Object {
                    "fontSize": 18,
                  },
                  false,
                  Object {
                    "flex": 1,
                    "fontWeight": "700",
                    "paddingHorizontal": 8,
                  },
                ]
              }
            >
              Lorem ipsum...
            </Text>
          </View>
          <RCTScrollView
            onScroll={[Function]}
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Dialog> when {visible} 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
        },
        Array [],
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minWidth": 320,
              "position": "absolute",
              "width": "100%",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 3,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 24,
              },
              Array [],
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "paddingHorizontal": 8,
                "paddingVertical": 8,
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
